web/src/compose_notifications.ts

```diff
@@ -245,6 +249,67 @@ export function notify_messages_outside_current_search(messages: Message[]): voi
     }
 }
 
+export function maybe_show_one_time_messages_fading_banner(): void {
+    const is_faded = $(".recipient_row").hasClass("message-fade");
+
+    if (!is_faded) {
+        return;
+    }
+
+    let recipient;
+    let narrow_url;
+    if (compose_state.get_message_type() === "stream") {
+        const stream_name = compose_state.stream_name();
+        const topic_name = compose_state.topic();
+        const stream_id = compose_state.stream_id();
+        assert(topic_name !== undefined);
+        assert(stream_id !== undefined);
+
+        recipient = `#${stream_name} > ${topic_name}`;
+
+        narrow_url = hash_util.by_stream_topic_url(stream_id, topic_name);
+    } else {
+        const emails_string = compose_state.private_message_recipient();
+        const user_ids_string = people.emails_strings_to_user_ids_string(emails_string);
+        assert(user_ids_string !== undefined);
+        const user_ids = people.user_ids_string_to_ids_array(user_ids_string);
+        const names = people.get_display_full_names(user_ids).sort();
+
+        recipient = util.format_array_as_list(names, "long", "conjunction");
+
+        if (names.length > 1) {
+            narrow_url = hash_util.huddle_with_url(user_ids_string);
+        } else {
+            narrow_url = hash_util.pm_with_url(emails_string);
+        }
+    }
+
+    const context = {
+        banner_type: compose_banner.INFO,
+        classname: compose_banner.CLASSNAMES.messages_fading,
+        button_text: $t({defaultMessage: "Got it"}),
+        hide_close_button: true,
+        recipient,
+        narrow_url,
+    };
+
+    const new_row_html = render_messages_fading_banner(context);
+
+    // If messages fading banner is already visible and the user
+    // is updating recipient, update the recipient in the banner.
+    const messages_fading_banner_selector = `#compose_banners .${CSS.escape(compose_banner.CLASSNAMES.messages_fading)}`;
+    if ($(messages_fading_banner_selector).length > 0) {
+        $(messages_fading_banner_selector).remove();
+        compose_banner.append_compose_banner_to_banner_list($(new_row_html), $("#compose_banners"));
+        return;
+    }
+
+    if (onboarding_steps.ONE_TIME_NOTICES_TO_DISPLAY.has("intro_messages_fading")) {
```

Comment:
I think we should be doing an early return if that `has` is not true, to avoid slowing down the app building all the HTML above that we're not going to use.

Reply:
Oh, but doing that would break the live-update support. Hmm.

Reply:
> if we decide we want to not mark it as "read" until you click "Got it"... which might be a better model.

https://chat.zulip.org/#narrow/stream/101-design/topic/explaining.20message.20fading/near/1781283

Reply:
We decided to go with the "Got it" approach.